METRON-1702: Reload a running job in the UI

merrimanr opened the pull request:
## Contributor Comments
This PR update the Pcap query panel so that a currently running query for a user is loaded when a page is refreshed.

### Changes Included

- Updated pcap-panel.component.ts to query for a running job, and if found, update the current status and start polling
- Refactored pcap-panel.component.ts by moving updating status and start polling into separate functions.  This should make the class more readable and is needed to update status after querying for a running job.
- Updated pcap-filters.component.ts to accept a PcapRequest as input and update filters to match the request made for the currently running job
- Added functions to pcap.service.ts to get the currently running job and the pcap request for that job
-  Fixed a bug in PcapServiceImpl.java where the wrong types were being used for some fields (also adjust unit tests to catch these errors in the future)
- Added tests for new code

### Testing
This has been tested in full dev. 

1. Navigate to the Pcap panel in the Alerts UI
2. Submit a query with filters and wait for progress to be > 0
3. Refresh the page
4. The page should look exactly the same (filters and all) and polling should continue until the job is complete

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?


### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

tiborm commented on metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts:
```diff
@@ -44,7 +44,23 @@ export class PcapPanelComponent {
   errorMsg: string;
 
-  constructor(private pcapService: PcapService) { }
+  constructor(private pcapService: PcapService ) { }
```
There is an extra space after the type. Is it intentional?

ruffle1986 commented on metron-interface/metron-alerts/src/app/pcap/pcap-panel/pcap-panel.component.ts:
```diff
@@ -15,22 +15,22 @@
   styleUrls: ['./pcap-panel.component.scss']
 })
-export class PcapPanelComponent {
+export class PcapPanelComponent implements OnInit {
```
I've already done it here in this commit: https://github.com/apache/metron/commit/14dcb2d90581835d8206c65918c24e8cb04bfd06
It's already merged back into the feature branch and it implements the destroy interface and I also refactored it a bit to make it more maintainable. We both touched the same lines so these conflicts have to be resolved before merging it back.

merrimanr commented:
Thanks @tiborm and @ruffle1986.  I merged in the latest feature branch which included the onDestroy code.  I also fixed a regression introduced by https://github.com/apache/metron/pull/1142 and added test to catch it in the future.  Let me know what you think.

sardell commented:
@merrimanr I reviewed and tested on full dev. I have the same thought as @ruffle1986 regarding the cancelling of the subscription. We should add that line inside the ngOnDestroy lifecycle hook. Besides that, tested on full dev and everything works as described. This is really great work and will significantly improve the user experience. +1 once the subscription cancellation is added.

ruffle1986 commented:
@merrimanr Looks good to me.

sardell commented:
+1. Tested on full dev.
